src/virtualMachines.js

```diff
@@ -1,30 +1,49 @@
 import VirtualMachine from './models/virtualMachine'
 
-
 class VirtualMachines {
-  props = VirtualMachine.props.join()
+  props = VirtualMachine.props.join()
   table = 'Orion.VIM.VirtualMachines'
 
   constructor(client) {
     this.client = client
   }
 
-  async query() {
-    const vms = await this.client.query(`
-      SELECT ${this.props}
-      FROM ${this.table}
-      ORDER BY VirtualMachineID
-    `)
+  async query(filter) {

```

Comment:
Can you refactor this function to get rid of duplicate code? We might use same approach later for `find` method.